docs/aws_temp_creds.md

```diff
@@ -0,0 +1,492 @@
+
+2\. In ZMS, create a policy authorizing your Athenz role members to
+assume the AWS role. When creating the policy the action must be
+`assume_aws_role`, choose the role as created in the previous step
+and specify the IAM role name as the resource.
+
+![Athenz Policy Setup](images/aws_athenz_policy_create.png)
+
+3\. Make sure your role contains all the service identities that
+may request temporary credentials for this role.
+
+4\. The changes will propagate from ZMS to ZTS running within AWS within
+2 minutes, and then you can use the ZTS Client library or its REST API to
+request temporary credentials.
+
+## Obtaining AWS Temporary Credentials
+
+We're going to look at how to obtain AWS Temporary Credentials with the
+use of the ZTS Client Library in Java and directly with ZTS
+Server's REST API.
+
+By default the AWS Temporary Credentials returned are valid only for 1 hour.
+If configured, it can be extended upto 12 hours - see [Expiration period](#expiration-period)
+section for full details. The client must refresh the credentials before they expire.
+
+### REST API
+
+To obtain a AWS Temporary Credentials, the application would use the
+following endpoint from ZTS Server running in AWS:
+
+```
+    GET /zts/v1/domain/{domainName}/role/{awsRoleName}/creds?durationSeconds={durationSeconds}&externalId={externalId}
+    durationSeconds and externalId query arguments are optional.
+```
+
+For example, with our setup as described above, `sherpa.api` is our
+service identity and we have retrieved the X.509 certificate for this
+service. The file locations are as follows:
+
+```
+    private-key: /var/lib/sia/keys/sherpa.api.key.pem
+    certificate: /var/lib/sia/certs/sherpa.api.cert.pem
+```
+
+The following request would retrieve AWS temporary session credentials for the
+`sherpa.property` domain's aws **IAM role** `s3-uploader`:
+
+```
+    curl --key /var/lib/sia/keys/sherpa.api.key.pem --cert /var/lib/sia/certs/sherpa.api.cert.pem https://your.zts/zts/v1/domain/sherpa.property/role/s3-uploader/creds
+```
+
+Note that the `awsRoleName` URL parameter is the name of the **IAM role** you want to assume,
+not the Athenz role created in ZMS.
+
+#### Expiration Period
+
+The following request would retrieve AWS temporary session credentials for the
+`sherpa.property` domain's aws iam role `s3-uploader` with duration set to 6
+hours (6 * 60 * 60 = 21600 seconds):
+
+```
+    curl --key /var/lib/sia/keys/sherpa.api.key.pem --cert /var/lib/sia/certs/sherpa.api.cert.pem https://your.zts/zts/v1/domain/sherpa.property/role/s3-uploader/creds?durationSeconds=21600
+```
+
+#### External ID Condition
+
+If we had configured our AssumeRole policy for the `s3-uploader` role with an external id of `hockey`
+then our request would be:
+
+```
+    curl --key /var/lib/sia/keys/sherpa.api.key.pem --cert /var/lib/sia/certs/sherpa.api.cert.pem https://your.zts/zts/v1/domain/sherpa.property/role/s3-uploader/creds?externalId=hockey
+```
+
+#### Case Sensitivity
```

Comment:
Removed entire section on Screwdriver deployment with aws-builder-creds (which isn't open source). Original text:
 ## Screwdriver Deployment Setup

 Screwdriver platform now supports Athenz X.509 Client certificates for
 all of its projects so you can use the REST API or the aws-builder-creds
 utility that the Athenz team provides to fetch temporary AWS credentials
 for your deployment task. You would still need to follow the complete
 steps defined in this document for your project.

 Within your Screwdriver job, you would fetch and install the aws-builder-creds
 utility:

 ```
 sudo yum-config-manager --add-repo https://edge.artifactory.yahoo.com:4443/artifactory/athens_rpms/athens_rpms.repo
 sudo yum install -y aws-builder-creds
 ```

 Once you have the utility installed, you can retrieve your aws temporary credentials (replace `athenz-domain` and `aws-iam-role-name` with their respective values for your environment):

 In SDv3, use

 ```
 aws-builder-creds -k /tokens/key -c /tokens/cert -d <athenz-domain> -r <aws-iam-role-name> -z zts.athenz.ouroath.com [-t <durationSeconds>] [-e <externalId>] > ~/.aws/credentials
 ```

 In SDv4, use

 ```
 aws-builder-creds -k /sd/tokens/key -c /sd/tokens/cert -d <athenz-domain> -r <aws-iam-role-name> -z zts.athenz.ouroath.com [-t <durationSeconds>] [-e <externalId>] > ~/.aws/credentials
 ```

 Finally, once you create your deployment role in Athenz in step 1
 in the [Athenz AWS Assume Role Configuration Setup](#athenz-aws-assume-role-configuration-setup)
 in the [Athenz AWS Assume Role Configuration Setup](#athenz-aws-assume-role-configuration-setup)
 section, you would add your Screwdriver Project Identity in the
 format `cd.screwdriver.project.sd<project-id>` to that role.

 You might consider using `aws_ansible_membership` SD v3 platform for your project.
 Even though it is developed by the membership team, it is fairly generic and can
 be used to deploy other services to AWS.